src/pages/dogs/[id]/forms/new.jsx

```diff
@@ -0,0 +1,125 @@
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import ConfirmCancelModal from "@/components/ConfirmCancelModal";
+import { ChevronLeftIcon } from "@heroicons/react/24/solid";
+import Link from "next/link";
+import {
+  MONTHLY_PLACED_FORM,
+  MONTHLY_UNPLACED_FORM,
+  VOLUNTEER_FORM,
+} from "@/utils/formConsts";
+
+export default function NewForm() {
```

Comment:
Make this a general Form component. You can have the component take in an action enum (view, edit, new) and based on that alter the inputs (disabled or not) + change the submit button logic

Reply:
done!
I refactored the newForm stuff into a general form component and added the view / add logic. [the edit logic still needs to be implemented]